sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSink.scala

```diff
@@ -51,7 +51,7 @@ object FileStreamSink extends Logging {
         val hdfsPath = new Path(singlePath)
         try {
           val fs = hdfsPath.getFileSystem(hadoopConf)
-          if (fs.isDirectory(hdfsPath)) {
+          if (fs.getFileStatus(hdfsPath).isDirectory) {
```

Comment:
In the original semantics, if `hdfsPath` does not exist, it will directly return `false`. In the new semantics, if `hdfsPath` does not exist, it will `throw FileNotFoundException`, execute the logic to `catch case NonFatal(e)`, and ultimately return `false`.

Reply:
Is the current log appropriate if `fs.getFileStatus(hdfsPath)` throws a `FileNotFoundException`?

Reply:
from "Assume no metadata directory. Error while looking for metadata directory in the path: $singlePath."

Because its parent directory does not exist in this case, it seems reasonable to log that the subdirectory does not exist.
Of course, if we want to add special explanations for this situation, it is also possible.